spring-pulsar-docs/src/main/antora/modules/ROOT/pages/reference/pulsar/transactions.adoc

```diff
@@ -0,0 +1,222 @@
+[[transactions]]
+= Transactions
+
+This section describes how Spring for Apache Pulsar supports transactions.
+
+[[overview]]
+== Overview
+Spring for Apache Pulsar transaction support is built upon the https://docs.spring.io/spring-framework/reference/data-access/transaction.html[transaction support] provided by Spring Framework.
+At a high-level, transactional resources are registered with a transaction manager which in turn handles the transactional state (commit, rollback, etc..) of the registered resources.
+
+Spring for Apache Pulsar provides the following:
+
+* `PulsarTransactionManager` - used with normal Spring transaction support (`@Transactional`, `TransactionTemplate`, etc)
+* Transactional `PulsarTemplate`
+* Transactional `@PulsarListener`
+* Transaction synchronization with other transaction managers
+
+NOTE: Transaction support has not been added to the Reactive components yet
+
+Transaction support is disabled by default.
+To enable support when using Spring Boot, simply set the `spring.pulsar.transaction.enabled` property.
+Further configuration options are outlined in each component section below.
+
+== Transactional Publishing with `PulsarTemplate`
+All send operations on a transactional `PulsarTemplate` look for an active transaction and enlist each send operation in the transaction (if one is found).
+
+=== Non-transactional use
+By default, a transactional `PulsarTemplate` can also be used for non-transactional operations.
+When an existing transaction is not found it will continue the send operation in a non-transactional fashion.
+However, if the template is configured to require transactions then any attempt to use the template outside the scope of a transaction results in an exception.
```

Comment:
What exception?

Reply:
Maybe it would be stated as:
> However, if the template is configured to require transactions and an existing transaction is not found, then it will not continue the send operation and instead will throw an exception.

Reply:
Just curious if we could say what type of exception is being thrown.

Reply:
I try not to do that in the docs because that can get out of sync easy because it is implementation specific (e.g. IllegalStateException today but maybe IllegalArgumentException next month).